python/ray/serve/_private/config.py

```diff
@@ -46,64 +32,8 @@ def _needs_pickle(deployment_language: DeploymentLanguage, is_cross_language: bo
         return False
 
 
-class DeploymentConfig(BaseModel):
-    """Internal datastructure wrapping config options for a deployment.
-
-    Args:
-        num_replicas (Optional[int]): The number of processes to start up that
-            handles requests to this deployment. Defaults to 1.
-        max_concurrent_queries (Optional[int]): The maximum number of queries
-            that is sent to a replica of this deployment without receiving
-            a response. Defaults to 100.
-        user_config (Optional[Any]): Arguments to pass to the reconfigure
-            method of the deployment. The reconfigure method is called if
-            user_config is not None. Must be JSON-serializable.
-        graceful_shutdown_wait_loop_s (Optional[float]): Duration
-            that deployment replicas wait until there is no more work to
-            be done before shutting down.
-        graceful_shutdown_timeout_s (Optional[float]):
-            Controller waits for this duration to forcefully kill the replica
-            for shutdown.
-        health_check_period_s (Optional[float]):
-            Frequency at which the controller health checks replicas.
-        health_check_timeout_s (Optional[float]):
-            Timeout that the controller waits for a response from the
-            replica's health check before marking it unhealthy.
-        user_configured_option_names (Set[str]):
-            The names of options manually configured by the user.
-    """
-
-    num_replicas: NonNegativeInt = Field(
-        default=1, update_type=DeploymentOptionUpdateType.LightWeight
-    )
-    max_concurrent_queries: Optional[int] = Field(
-        default=None, update_type=DeploymentOptionUpdateType.NeedsReconfigure
-    )
-    user_config: Any = Field(
-        default=None, update_type=DeploymentOptionUpdateType.NeedsActorReconfigure
-    )
-
-    graceful_shutdown_timeout_s: NonNegativeFloat = Field(
-        default=DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT_S,
-        update_type=DeploymentOptionUpdateType.NeedsReconfigure,
-    )
-    graceful_shutdown_wait_loop_s: NonNegativeFloat = Field(
-        default=DEFAULT_GRACEFUL_SHUTDOWN_WAIT_LOOP_S,
-        update_type=DeploymentOptionUpdateType.NeedsActorReconfigure,
-    )
-
-    health_check_period_s: PositiveFloat = Field(
-        default=DEFAULT_HEALTH_CHECK_PERIOD_S,
-        update_type=DeploymentOptionUpdateType.NeedsReconfigure,
-    )
-    health_check_timeout_s: PositiveFloat = Field(
-        default=DEFAULT_HEALTH_CHECK_TIMEOUT_S,
-        update_type=DeploymentOptionUpdateType.NeedsReconfigure,
-    )
-
-    autoscaling_config: Optional[AutoscalingConfig] = Field(
-        default=None, update_type=DeploymentOptionUpdateType.LightWeight
-    )
+class InternalDeploymentConfig(BaseDeploymentModel):
```

Comment:
Synced offline, will potentially push this off to a later PR to avoid adding to the LOC changes in this PR. Thanks @shrekris-anyscale for the suggestion!